src/components/PostPageDetail/index.js

```diff
@@ -585,7 +585,7 @@ const PostPageDetailIdComponent = (props) => {
             isBackButton={true}
             source={SOURCE_PDP}
             height={getHeightHeader()}
-            isFollow={item?.is_following_target || parentData?.is_following_target}
+            isFollow={item?.is_following_target}
             onPressFollUnFoll={() => {
               followUnfollow(item);
               setItem({...item, is_following_target: !item?.is_following_target});
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onPressFollUnFoll` function has been modified to directly update the `item` state. This is an optimistic UI update, assuming the `followUnfollow(item)` function will succeed. While this can improve perceived performance by instantly reflecting changes in the UI, it may lead to inconsistencies if the `followUnfollow(item)` operation fails. Consider adding error handling to revert the state change in case of failure.
```javascript
onPressFollUnFoll={() => {
  followUnfollow(item).catch(() => {
    // Revert the state change upon failure
    setItem({...item, is_following_target: item?.is_following_target});
  });
  setItem({...item, is_following_target: !item?.is_following_target});
}}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->